lib/process-services-cloud/src/lib/task/task-list/components/base-task-list-cloud.component.ts

```diff
@@ -0,0 +1,276 @@
+    @Input()
+    showContextMenu: boolean = false;
+
+    /** Emitted before the context menu is displayed for a row. */
+    @Output()
+    showRowContextMenu = new EventEmitter<DataCellEvent>();
+
+    /** Emitted before the actions menu is displayed for a row. */
+    @Output()
+    showRowActionsMenu = new EventEmitter<DataCellEvent>();
+
+    /** Emitted when the user executes a row action. */
+    @Output()
+    executeRowAction = new EventEmitter<DataRowActionEvent>();
+
+    /** Emitted when a task in the list is clicked */
+    @Output()
+    rowClick: EventEmitter<string> = new EventEmitter<string>();
+
+    /** Emitted when rows are selected/unselected */
+    @Output()
+    rowsSelected: EventEmitter<any[]> = new EventEmitter<any[]>();
+
+    /** Emitted when the task list is loaded */
+    @Output()
+    success: EventEmitter<any> = new EventEmitter<any>();
+
+    /** Emitted when an error occurs. */
+    @Output()
+    error: EventEmitter<any> = new EventEmitter<any>();
+
+    pagination: BehaviorSubject<PaginationModel>;
+
+    requestNode: TaskQueryCloudRequestModel;
+    rows: any[] = [];
+    size: number;
+    skipCount: number = 0;
+    currentInstanceId: any;
+    isLoading = true;
+    selectedInstances: any[];
+    formattedSorting: any[];
+    private defaultSorting = { key: 'startDate', direction: 'desc' };
+
+    private onDestroy$ = new Subject<boolean>();
+
+    constructor(appConfigService: AppConfigService,
+                private userPreferences: UserPreferencesService,
+                presetKey: string) {
+        super(appConfigService, presetKey, taskPresetsCloudDefaultModel);
+        this.size = userPreferences.paginationSize;
+
+        this.pagination = new BehaviorSubject<PaginationModel>(<PaginationModel> {
+            maxItems: this.size,
+            skipCount: 0,
+            totalItems: 0
+        });
+
+    }
+
+    ngOnInit() {
+        this.userPreferences
+            .select(UserPreferenceValues.PaginationSize)
+            .pipe(takeUntil(this.onDestroy$))
+            .subscribe(pageSize => this.size = pageSize);
+    }
+
+    ngOnChanges(changes: SimpleChanges) {
+        if (this.isPropertyChanged(changes, 'sorting')) {
+            this.formatSorting(changes['sorting'].currentValue);
+        }
+        if (this.isAnyPropertyChanged(changes)) {
+            this.reload();
+        }
+    }
+
+    ngOnDestroy() {
+        this.onDestroy$.next(true);
+        this.onDestroy$.complete();
+    }
+
+    ngAfterContentInit() {
+        this.createDatatableSchema();
+    }
+
+    getCurrentId(): string {
+        return this.currentInstanceId;
+    }
+
+    private isAnyPropertyChanged(changes: SimpleChanges): boolean {
+        for (const property in changes) {
+            if (this.isPropertyChanged(changes, property)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private isPropertyChanged(changes: SimpleChanges, property: string): boolean {
```

Comment:
not needed at all, if the ngOnChanges is fired, you already have something that is changed, no need doing the secondary job of Angular